tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -20,16 +22,33 @@ internal class LambdaCommon
         private const string DefaultJson = "{}";
         private const double ServerlessMaxWaitingFlushTime = 3;
 
-        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder)
+        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder, IDictionary<string, string> context)
         {
             var json = SerializeObject(payload);
-            NotifyExtensionStart(requestBuilder, json);
+            NotifyExtensionStart(requestBuilder, json, context ?? new Dictionary<string, string>());
```

Comment:
If possible, we should avoid the heap allocation here (`new Dictionary`) by either letting the the `null` pass through to `NotifyExtensionStart()` (and handle it accordingly), or caching a single empty readonly dictionary that can reused in case `context` is `null`. The second option probably keep the code cleaner (don't have to add null checks everywhere), but only works if the dictionary is never modified (in which case you may want to use `IReadOnlyDictionary`, anyway).

Reply:
I'm now propagating the null value down to where this variable is used (`WriteRequestHeaders`) where it is handled